presentation/src/main/java/com/dcs/presentation/trend/TrendRoute.kt

```diff
@@ -0,0 +1,213 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.ScrollableTabRow
+import androidx.compose.material3.Tab
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.paging.PagingData
+import androidx.paging.compose.collectAsLazyPagingItems
+import com.dcs.presentation.R
+import com.dcs.presentation.common.ErrorScreen
+import com.dcs.presentation.common.LoadingScreen
+import com.dcs.presentation.core.model.MovieItemUiState
+import com.dcs.presentation.core.state.MoviePopularType
+import com.dcs.presentation.core.state.MovieTrendType
+import com.dcs.presentation.core.state.UiState
+import kotlinx.coroutines.flow.StateFlow
+
+@Composable
+fun TrendRoute(
+    modifier: Modifier = Modifier,
+    viewModel: TrendViewModel = hiltViewModel()
+) {
+
+    Scaffold { innerPadding ->
+        val scrollState = rememberScrollState()
+
+        Column(
+            modifier = modifier
+                .fillMaxSize()
+                .padding(innerPadding)
+                .verticalScroll(scrollState)
+        ) {
+            TrendMoviePanel(
+                title = stringResource(id = R.string.movie_trend_title_trend),
+                tabs = MovieTrendType.entries.map { it.toUiString() },
+                movies = viewModel.moviesByTrending
+            ) { tabIndex ->
+                viewModel.updateMovieTrendType(MovieTrendType.entries[tabIndex])
+            }
+
+            TrendMoviePanel(
+                title = stringResource(id = R.string.movie_trend_title_popular),
+                tabs = MoviePopularType.entries.map { it.toUiString() },
+                movies = viewModel.moviesByPopular
+            ) { tabIndex ->
+                viewModel.updateMoviePopularType(MoviePopularType.entries[tabIndex])
+            }
+
+            TrendMoviePanel(
+                title = stringResource(id = R.string.movie_trend_title_upcoming),
+                movies = viewModel.moviesByUpcoming
+            )
```

Comment:
Section 레벨의 함수에서는 어떤 영역이라고 명시적으로 알 수 있게끔 함수를 선언해서 하위 컴퍼넌트를 선언하는게 컴포즈 함수 읽기 쉽기 때문에 컴포저블 함수를 한번 더 감싸는게 좋을 것 같아요!
```suggestion
            TopRatedMovies()
            PopularMovies()
            TrendMovies()
      .....
      fun TopRatedMovies() { 
            TrendMoviePanel(
                title = stringResource(id = R.string.movie_trend_title_trend),
                tabs = MovieTrendType.entries.map { it.toUiString() },
                movies = viewModel.moviesByTrending
            ) { tabIndex ->
                viewModel.updateMovieTrendType(MovieTrendType.entries[tabIndex])
            }
      }
      ...
```

Reply:
적용 완료. 읽기 좋게 만들기 위해 컴포저블 함수로 한번 더 랩핑하는게 맞는 방식인가??

Reply:
네 맞아요. 주석으로 달아도 되긴 하는데 나중에 다른 하나의 화면에 컴포넌트가 추가 된다고 했을 때 breaking changes가 범위를 제한 하는데 이유도 있긴하죠.